From 3ca2d2d4aef910663785fe232d7d2127a22d8a84 Mon Sep 17 00:00:00 2001
From: Daniel Sabogal <dsabogalcc@gmail.com>
Date: Sat, 17 Sep 2016 12:05:45 -0400
Subject: [PATCH 18/25] fix clock_nanosleep error case

posix requires that EINVAL be returned if the first parameter specifies
the cpu-time clock of the calling thread (CLOCK_THREAD_CPUTIME_ID).
linux returns ENOTSUP instead so we handle this.
---
 src/time/clock_nanosleep.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/time/clock_nanosleep.c b/src/time/clock_nanosleep.c
index ec87b9e..9e4d9f1 100644
--- a/src/time/clock_nanosleep.c
+++ b/src/time/clock_nanosleep.c
@@ -1,8 +1,10 @@
 #include <time.h>
+#include <errno.h>
 #include "syscall.h"
 #include "libc.h"
 
 int clock_nanosleep(clockid_t clk, int flags, const struct timespec *req, struct timespec *rem)
 {
-	return -__syscall_cp(SYS_clock_nanosleep, clk, flags, req, rem);
+	int r = -__syscall_cp(SYS_clock_nanosleep, clk, flags, req, rem);
+	return clk == CLOCK_THREAD_CPUTIME_ID ? EINVAL : r;
 }
-- 
2.10.1

